src/main/java/com/o19s/es/explore/PostingsExplorerQuery.java

```diff
@@ -189,4 +197,67 @@ public float getMaxScore(int upTo) throws IOException {
             return Float.POSITIVE_INFINITY;
         }
     }
+
+    static class TPScorer extends PostingsExplorerScorer {
+        TPScorer(Weight weight, PostingsEnum postingsEnum) {
+            super(weight, postingsEnum);
+        }
+        final float NORMALIZED_TP = 5000.0f;
```

Comment:
Slightly nervous about this constant.  Will need to review the logic after gaining a better understanding of what normalized average is accomplishing.

Reply:
This is a default weight to penalize documents without term second position.

Reply:
@MaquisW may explain it the better way to @worleydl why we created this feature.

Reply:
That's exactly what @rtancman explained. In this case, `norma_avg` is computed only with the first two occurrences of any query term, the idea is that this feature demonstrates how far, on average, the first two occurrences of any query term are in the document. With that in mind, it is possible that documents that have only one occurrence may have an advantage in calculating this average, so we simulate a fake second occurrence. Ideally, this fake occurrence should penalize the calculation of the average of the first 2 occurrence positions in this document, considering that none of the terms of the query occurred a second time, which must be bad for the document's score. One way to penalize is to simulate that the fake occurrence happens at the end of the document, so `NORMALIZED_TP` (which needs a better name, by the way) could have a value like `size_of_the_document + 1`, but small documents could continue to have an advantage (and we also had trouble getting the document size at this point), so we assume a high value like 5000. We are still experimenting and studying the engineering possibilities of this feature, and we saw that all these calculations still have some issues. We appreciate the feedback, soon we will update the PR with more information and clearer documentation. Thank you!

Reply:
Thanks for the explanation. I can see the value of the first N occurrences feature, and I also agree there is experimentation to be done about how to normalize for document length and what to do with <N occurrences. 

Because this feature is still being developed, would you be okay if we only include `min`, `max` and `avg` in the PR for now?

We just want to make sure these the plugin supports features that a broadly useful for a variety of Elasticsearch users/cases.